### PR TITLE
Add raytrilium extension

### PR DESCRIPTION
**Pull request by jamtur01**
## Description

Adds an extension that creates notes in [Trilium](https://github.com/zadam/trilium) from Raycast. Makes use of Trilum's ETAPI and requires the user generate an API key and put it into the Raycast configuration.

## Screencast

screenshot

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/raytrilium/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Submit" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.Description text="Create a new note in Trilium." />
+      <Form.TextField id="title" title="Note title" placeholder="Enter the note's title" />
```
Comment (pernielsentikaer) [inline extensions/raytrilium/src/index.tsx:123]:
```suggestion
      <Form.TextField id="title" title="Note Title" placeholder="Enter the note's title" />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- You can use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) which handles states, errors and validation

- When using wrong credentials I still get a success toast

raytrilium 2024-01-31 at 10 44 01


- Consider `popToRoot()` after filling the form or clear it so it can be used again 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Is this ready for another review?

**Comment by jamtur01** [comment]
@pernielsentikaer Yes, although the API seems to work for me locally with or without a correct token so I am poking into what's going on there.

**Comment by pernielsentikaer** [comment]
Converting to a draft for now, just ping me then 🙂

**Pull request closed without merging**